### ISSUE TITLE
vmstoragre ingest very slowly after IndexDB rotation

### ISSUE THREAD

**Issue by jianglinjian** [VictoriaMetrics/VictoriaMetrics#2113]
**Describe the bug**
Because tsidCache was resetted after indexDB rotation, data insertion would slow down about 4x for a short period of time.

**To Reproduce**
next IndexDB rotation 

**Expected behavior**
Insertion performance should not jitter (at least not significantly) after indexDB rotation.

**Logs**
```
2022-01-28T04:00:00.000Z	info	VictoriaMetrics/lib/mergeset/table.go:260	opening table "/vm-data/indexdb/16BA385755E883D4"...
2022-01-28T04:00:00.004Z	info	VictoriaMetrics/lib/mergeset/table.go:295	table "/vm-data/indexdb/16BA385755E883D4" has been opened in 0.004 seconds; partsCount: 0; blocksCount: 0, itemsCount: 0; sizeBytes: 0
2022-01-28T04:00:00.005Z	info	VictoriaMetrics/lib/mergeset/table.go:332	waiting for raw items flusher to stop on "/vm-data/indexdb/16BA385755E883D2"...
2022-01-28T04:00:00.005Z	info	VictoriaMetrics/lib/mergeset/table.go:335	raw items flusher stopped in 0.000 seconds on "/vm-data/indexdb/16BA385755E883D2"
2022-01-28T04:00:00.005Z	info	VictoriaMetrics/lib/mergeset/table.go:337	waiting for converters to stop on "/vm-data/indexdb/16BA385755E883D2"...
2022-01-28T04:00:00.005Z	info	VictoriaMetrics/lib/mergeset/table.go:340	converters stopped in 0.000 seconds on "/vm-data/indexdb/16BA385755E883D2"
2022-01-28T04:00:00.005Z	info	VictoriaMetrics/lib/mergeset/table.go:342	waiting for part mergers to stop on "/vm-data/indexdb/16BA385755E883D2"...
2022-01-28T04:00:00.005Z	info	VictoriaMetrics/lib/mergeset/table.go:345	part mergers stopped in 0.000 seconds on "/vm-data/indexdb/16BA385755E883D2"
2022-01-28T04:00:00.005Z	info	VictoriaMetrics/lib/mergeset/table.go:347	flushing inmemory parts to files on "/vm-data/indexdb/16BA385755E883D2"...
2022-01-28T04:00:00.005Z	info	VictoriaMetrics/lib/mergeset/table.go:371	0 inmemory parts have been flushed to files in 0.000 seconds on "/vm-data/indexdb/16BA385755E883D2"
2022-01-28T04:00:00.005Z	info	VictoriaMetrics/lib/mergeset/table.go:373	waiting for flush callback worker to stop on "/vm-data/indexdb/16BA385755E883D2"...
2022-01-28T04:00:00.005Z	info	VictoriaMetrics/lib/mergeset/table.go:376	flush callback worker stopped in 0.000 seconds on "/vm-data/indexdb/16BA385755E883D2"
2022-01-28T04:00:00.071Z	info	VictoriaMetrics/lib/storage/index_db.go:268	dropping indexDB "/vm-data/indexdb/16BA385755E883D2"
2022-01-28T04:00:00.634Z	info	VictoriaMetrics/lib/storage/index_db.go:270	indexDB "/vm-data/indexdb/16BA385755E883D2" has been dropped
2022-01-28T04:00:00.889Z	info	VictoriaMetrics/lib/storage/storage.go:983	saving MetricName->TSID cache to "/vm-data/cache/metricName_tsid"...
2022-01-28T04:00:00.974Z	info	VictoriaMetrics/lib/storage/storage.go:990	saved MetricName->TSID cache to "/vm-data/cache/metricName_tsid" in 0.085 seconds; entriesCount: 15560; sizeBytes: 31719424
2022-01-28T04:01:46.730Z	info	VictoriaMetrics/app/vmstorage/transport/server.go:164	accepted vmselect conn from 10.249.3.195:16990
2022-01-28T04:01:46.730Z	info	VictoriaMetrics/app/vmstorage/transport/server.go:164	accepted vmselect conn from 10.249.3.195:16992
2022-01-28T04:01:46.730Z	info	VictoriaMetrics/app/vmstorage/transport/server.go:164	accepted vmselect conn from 10.249.3.195:16994
2022-01-28T04:01:46.730Z	info	VictoriaMetrics/app/vmstorage/transport/server.go:164	accepted vmselect conn from 10.249.3.195:16996
2022-01-28T04:01:46.730Z	info	VictoriaMetrics/app/vmstorage/transport/server.go:164	accepted vmselect conn from 10.249.3.195:16998
2022-01-28T04:01:46.730Z	info	VictoriaMetrics/app/vmstorage/transport/server.go:164	accepted vmselect conn from 10.249.3.195:17000
2022-01-28T04:01:46.786Z	info	VictoriaMetrics/app/vmstorage/transport/server.go:207	processing vmselect conn from 10.249.3.195:16996
2022-01-28T04:01:46.789Z	info	VictoriaMetrics/app/vmstorage/transport/server.go:207	processing vmselect conn from 10.249.3.195:16994
2022-01-28T04:01:46.797Z	info	VictoriaMetrics/app/vmstorage/transport/server.go:207	processing vmselect conn from 10.249.3.195:17000
2022-01-28T04:01:46.798Z	info	VictoriaMetrics/app/vmstorage/transport/server.go:207	processing vmselect conn from 10.249.3.195:16998
2022-01-28T04:01:46.803Z	info	VictoriaMetrics/app/vmstorage/transport/server.go:207	processing vmselect conn from 10.249.3.195:16992
2022-01-28T04:01:46.806Z	info	VictoriaMetrics/app/vmstorage/transport/server.go:207	processing vmselect conn from 10.249.3.195:16990
```

**Screenshots**
<img width="2498" alt="截屏2022-01-28 下午4 02 22" src="https://user-images.githubusercontent.com/26756408/151509727-c50d7e71-847d-40f1-9ae2-85381468b0bc.png">

<img width="1255" alt="截屏2022-01-28 下午1 13 49" src="https://user-images.githubusercontent.com/26756408/151509784-f8c7a211-0b4f-44c1-93d8-e12bb9fdf8b4.png">

<img width="1252" alt="截屏2022-01-28 下午3 59 55" src="https://user-images.githubusercontent.com/26756408/151509819-d49db707-f844-4530-b9ba-b1daecc342ba.png">

<img width="2488" alt="截屏2022-01-28 下午4 24 14" src="https://user-images.githubusercontent.com/26756408/151512762-9e248f36-926b-41be-a0d9-9f6c4f42786c.png">

**Version**
victoriametrics/vminsert:v1.64.1-cluster
victoriametrics/vmstorage:v1.64.1-cluster
victoriametrics/vmselect:v1.64.1-cluster

**Comment by valyala** [comment]
@jianglinjian , thanks for filing the bugreport!

Closing the issue as a duplicate for https://github.com/VictoriaMetrics/VictoriaMetrics/issues/1401 . We are going to address this issue in the next few weeks. See also https://github.com/VictoriaMetrics/VictoriaMetrics/pull/2139#issuecomment-1031358743 .

**Comment by gavenpeng** [comment]
Dear @jianglinjian , the problem is resolved or not , we use 1.75.0 also happened , not flush tsid cache while indexdb rotation,but problem is already exist

**Comment by jianglinjian** [comment]
@gavenpeng Due to the limitations of the index data structure, it is inevitable that there will be performance fluctuations during retention. The optimization in version v1.73.0 is to smooth out the peak within an hour after the retention. You can follow this [issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/1401), Valyala is continuously optimizing it.
